ONSdigital/dp-dataset-api
# Feature/import task validation

### What

Put endpoint to update Import tasks on an instance did not have enough validation, even though the data would never be corrupted, the response messages and status were not correct.

New validation models force status of bad request (400) when the request body does not contain the right structure or data.

### How to review

Check changes are logical and make sense. Make sure unit tests pass. Run acceptance tests by doing the following:

1) Run `dp-auth-api-stub` on `cmd-develop` with `make debug`
2) Run `dp-dataset-api` on `feature/import-task-validation` with `make acceptance-publishing`
3) Run acceptance tests in `dp-api-tests` on `feature/extend-missing-dataset-tests` with `cd publishing/datasetAPI/; go test ./...` 

### Who can review

Anyone
